File: app/src/main/java/com/imangazalievm/bubbble/data/users/FollowersRepository.java
package com.imangazalievm.bubbble.data.users;

import com.imangazalievm.bubbble.data.global.network.DribbbleApiService;
import com.imangazalievm.bubbble.domain.global.models.Follow;
import com.imangazalievm.bubbble.domain.global.models.UserFollowersRequestParams;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class FollowersRepository {

    private DribbbleApiService dribbbleApiService;

    @Inject
    public FollowersRepository(DribbbleApiService dribbbleApiService) {
        this.dribbbleApiService = dribbbleApiService;
    }

    public Single<List<Follow>> getUserFollowers(UserFollowersRequestParams requestParams) {
        return dribbbleApiService.getUserFollowers(requestParams.getUserId(), requestParams.getPage(), requestParams.getPageSize());
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/data/users/FollowersRepository.kt
package com.imangazalievm.bubbble.data.users

import com.imangazalievm.bubbble.data.global.network.DribbbleApi
import com.imangazalievm.bubbble.domain.global.models.Follow
import com.imangazalievm.bubbble.domain.global.models.UserFollowersRequestParams
import javax.inject.Inject

class FollowersRepository @Inject constructor(
    private val dribbbleApi: DribbbleApi
) {

    suspend fun getUserFollowers(requestParams: UserFollowersRequestParams): List<Follow> {
        return dribbbleApi.getUserFollowers(
            requestParams.userId,
            requestParams.page,
            requestParams.pageSize
        )
    }

}